apprating/src/main/java/com/mindera/skeletoid/apprating/callbacks/AppRatingDialogCallback.kt

```diff
@@ -0,0 +1,8 @@
+package com.mindera.skeletoid.apprating.callbacks
+
+/**
+ * Callback to prompt the rating dialog. To use if intended to create a custom dialog
+ */
+interface AppRatingDialogCallback {
+    fun showRatingDialog()
+}
```

Comment:
I'm thinking, wouldn't it be good to have all the callback there, like "never" and "not now" so that it can also be handled if someone created a custom dialog? like, if someone wants to display some sort of "you suck" toast or sth?

Reply:
This callback is only used whenever it's intended that the app itself is responsible for showing the dialog.
So when the Activity calls `AppRatingInitializer.promptDialog(context, callback: AppRatingDialogCallback)`, skeletoid validates that it can show the dialog and calls `callback.showRatingDialog()` , and then the activity handles the initialization of the dialog. On the other hand, imagine that the app intends to use the our default dialog (to be implemented) the activity calls `AppRatingInitializer.promptDialog(context, callback: AppRatingDialogResponseCallback)` , skeletoid validates the conditions, shows the dialog and if the user clicks "Rate Now" it calls `callback.onRateNowClick()` . 

Were you talking in this last case about adding the other responses like: `onRateLaterClick` and `OnNeverRateClick()` instead of having only the `onRateNowClick`?

Reply:
ah, ok, get it now :)
yeah, it could be a good idea to have it actually I think :)